apps/zbugs/shared/mutators.ts

```diff
@@ -179,6 +246,49 @@ export function createMutators(authData: AuthData | undefined) {
       creatorID,
       created,
     });
+
+    // subscribe to notifications if the user emojis the issue itself
+    if (subjectType === 'issue') {
+      await updateIssueNotification(tx, {
+        userID: creatorID,
+        issueID: subjectID,
+        subscribed: 'subscribe',
+        created,
+      });
+    }
+  }
+
+  async function updateIssueNotification(
+    tx: Transaction<typeof schema, unknown>,
+    {
+      userID,
+      issueID,
+      subscribed,
+      created,
+    }: {
+      userID: string;
+      issueID: string;
+      subscribed: NotificationType;
+      created: number;
+    },
+  ) {
+    await assertUserCanSeeIssue(tx, userID, issueID);
+
+    if (subscribed === 'subscribe') {
+      await tx.mutate.issueNotifications.upsert({
+        userID,
+        issueID,
+        subscribed: true,
```

Comment:
This will update the user to subscribed even if they have already previously unsubscribed. I don't think that's what we want right? I think this function should only insert a record and set it to true if there isn't already a record set to false. For the unsubscribe case, upsert is correct.

Reply:
For the case where a user specifically sets their notification pref, using the UI then of course what it's doing now is the right thing.